packages/core/docs/migrate/2.3.0-rc.1/overview.md

```diff
@@ -0,0 +1,151 @@
+
+Breaking changes are related to the introduction of our new API middleware, an overhaul of the checkout, and updates to `@vue-storefront/cache` library.
+
+If you are an integrator, please also refer to the [Integratiors](./integrators.md) page.
+
+## API middleware
+
+If you watched the talk of our CTO at Vue.js Amsterdam you may be already aware of this change. If not, you can find his slides [here](https://slides.com/filiprakowski/vuejs-amsterdam-2021).
+
+In short, it's meant to reduce the amount of code we send to the browser and perform as many operations on the backend as possible. This greatly improves security and frontend performance but also gives you more control of what is sent to the 3rd party providers.
+
+Before, every request sent to the providers was made directly from the browser (besides the first visit on the page which is Server-Side Rendered), which meant that it had to have all the logic required to create and send the request and to process the response. In some cases browser had to load heavy libraries like GraphQL and have access to sensitive information, such as API keys.
+This meant that Vue Storefront would not scale as well as we would like.
+
+To fix these issues we moved most of the logic to the new API middleware. It's an Express server that lives beside Nuxt.js. It can be run in parallel on the same server and separately on a different server, for better scalability.
+Now, instead of making API calls directly to the provider, the browser will request data from the API middleware.
+
+### Changes
+
+1. Create a new configuration files in the root of your Nuxt project named `middleware.config.js`. Content is dependant on the integration you use. As an example we will use our `commercetools` integration:
+
+```javascript
+// middleware.config.js
+module.exports = {
+  integrations: {
+    ct: {
+      location: '@vue-storefront/commercetools-api/server',
+      configuration: {
+        api: {
+          uri: '',
+          authHost: '',
+          projectKey: '',
+          clientId: '',
+          clientSecret: '',
+          scopes: []
+        },
+        currency: '',
+        country: ''
+      }
+    }
+  }
+};
+
+```
+
+2. Add `'@vue-storefront/middleware/nuxt'` to the `modules` in `nuxt.config.js`.
+
+:::warning Be careful
+Make sure this package is added to the `modules` array, not `buildModules`.
+:::
+
+```javascript
+// nuxt.config.js
+export default {
+  modules: [
+    '@vue-storefront/middleware/nuxt'
+  ]
+};
+```
+
+3. Update custom queries.
+
+Previously, custom queries were callbacks passed as an additional argument in most of our composable methods. Because we moved API communication to the backend, it's no longer possible to pass callbacks from UI (frontend) to `api-client` (backend).
+
+From now on, custom queries are registered in `middleware.config.js` (with the same signature as before), and used on the frontend by providing their unique name:
```

Comment:
`From now on,` to `From now,`?

Reply:
I think `From now on` is correct - https://hinative.com/en-US/questions/40493